packages/labs/testing/src/web-test-runner/worker/worker.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {parentPort} from 'worker_threads';
+import {render} from '@lit-labs/ssr/lib/render-with-global-dom-shim.js';
+
+import type {
+  InitializeResponse,
+  RenderResponse,
+  RequestMessage,
+} from './types.js';
+
+if (parentPort === null) {
+  throw new Error('worker.js must only be run in a worker thread');
+}
+
+let initialized = false;
+
+parentPort.on('message', async (message: RequestMessage) => {
+  switch (message.type) {
+    case 'initialize-request': {
```

Comment:
Good point. This structure was necessary in the eleventy plugin because it needed multiple render requests with a single initialize. Here, we start a new worker thread for each command invocation so this could be a lot more simplified.